File: apps/api/app/schemas/agent.py
"""Agent types and state schemas."""

from typing import Literal, Optional

from pydantic import BaseModel, Field

AgentId = Literal["analyzer", "source", "perspective", "socrates"]
AgentStatus = Literal["idle", "thinking", "searching", "analyzing", "done", "error"]


class AgentState(BaseModel):
    """Represents the current state of an AI agent."""

    agent_id: AgentId = Field(..., description="Unique identifier for the agent")
    status: AgentStatus = Field(..., description="Current status of the agent")
    message: Optional[str] = Field(None, description="Status message or description")
    progress: Optional[int] = Field(
        None, ge=0, le=100, description="Progress percentage (0-100)"
    )

File: apps/web/lib/types/agent.ts
import type { BiasType } from './analysis';

export type AgentId = 'analyzer' | 'source' | 'perspective' | 'socrates';

export type AgentStatus =
  | 'idle'
  | 'thinking'
  | 'searching'
  | 'analyzing'
  | 'done'
  | 'error';

export interface AgentState {
  id: AgentId;
  status: AgentStatus;
  message?: string;
  progress?: number;
  result?: unknown;
  error?: string;
}

export interface AnalyzerOutput {
  claims: Claim[];
  logicStructure: string;
  detectedBiases: DetectedBias[];
  agentInstructions: {
    sourceVerifier: SourceVerifierInstruction;
    perspectiveExplorer: PerspectiveInstruction;
  };
}

export interface Claim {
  id: number;
  text: string;
  evidence: string;
  sources: string[];
}

export interface DetectedBias {
  type: BiasType;
  confidence: number;
  example: string;
}

export interface SourceVerifierInstruction {
  sources: string[];
  checkFor: string[];
}

export interface PerspectiveInstruction {
  topic: string;
  keywords: string[];
}
